tests/staking/observable/createWithdrawTx.test.ts

```diff
@@ -0,0 +1,169 @@
+    publicKeyNoCoordHex: keys.publicKeyNoCoord,
+    publicKeyWithCoord: keys.publicKey,
+  }
+  const observableStaking = new ObservableStaking(network, stakerInfo);
+  const unbondingTx = observableStaking.createUnbondingTransaction(
+    params,
+    delegation,
+  ).psbt.signAllInputs(keys.keyPair).finalizeAllInputs().extractTransaction();
+
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  describe("Create withdraw early unbonded transaction", () => {
+    it(`${networkName} should throw an error if delegation input is invalid`, async () => {
+      // Staking tx height is less than activation height
+      const invalidStakingTxHeight = dataGenerator.getRandomIntegerBetween(0, params.activationHeight);
+      const invalidStakingTx = {
+        ...observableStakingTransaction,
+        startHeight: invalidStakingTxHeight,
+      }
+      const invalidDelegation = {
+        ...delegation,
+        stakingTx: invalidStakingTx,
+      }
+      const staking = new ObservableStaking(network, stakerInfo);
+      jest.spyOn(observableStakingUtils, "validateAndDecodeDelegationInputs").mockImplementationOnce(() => {
+        throw new Error("Fail to validate delegation inputs");
+      });
+      expect(() => staking.createWithdrawEarlyUnbondedTransaction(
+        params, invalidDelegation, unbondingTx, feeRate,
+      )).toThrow("Fail to validate delegation inputs");
+    });
+
+    it(`${networkName} should throw an error if fail to build scripts`, async () => {
+      jest.spyOn(stakingScript, "StakingScriptData").mockImplementation(() => {
+        throw new Error("withdraw early unbonded delegation build script error");
+      });
+      
+      const observableStaking = new ObservableStaking(network, stakerInfo);
+      expect(() => observableStaking.createWithdrawEarlyUnbondedTransaction(
+        params,
+        delegation,
+        unbondingTx,
+        feeRate,
+      )).toThrow("withdraw early unbonded delegation build script error");
+    });
+
+    it(`${networkName} should throw an error if fail to build early unbonded withdraw tx`, async () => {
+      jest.spyOn(staking, "withdrawEarlyUnbondedTransaction").mockImplementation(() => {
+        throw new Error("fail to build withdraw tx");
+      });
+      const observableStaking = new ObservableStaking(network, stakerInfo);
+
+      expect(() => observableStaking.createWithdrawEarlyUnbondedTransaction(
+        params,
+        delegation,
+        unbondingTx,
+        feeRate,
+      )).toThrow("fail to build withdraw tx");
+    });
+
+    it(`${networkName} should create withdraw early unbonded transaction`, async () => {
+      const withdrawTx = observableStaking.createWithdrawEarlyUnbondedTransaction(
+        params,
+        delegation,
+        unbondingTx,
+        feeRate,
+      );
+      expect(withdrawTx).toBeDefined();
```

Comment:
Shouldn't we check the properties of the withdrawal tx? For example inputs/outputs, version, etc.